socorro/external/boto/crashstorage.py

```diff
@@ -272,6 +276,70 @@ class BotoS3CrashStorage(BotoCrashStorage):
 
 
 #==============================================================================
+class TelemetryBotoS3CrashStorage(BotoS3CrashStorage):
+    """S3 crash storage class for sending a subset of the processed crash
+    but reduced to only include the files in the processed crash
+    JSON Schema."""
+
+    required_config = Namespace()
+    required_config.resource_class = change_default(
+        BotoCrashStorage,
+        'resource_class',
+        'socorro.external.boto.connection_context.RegionalS3ConnectionContext'
+    )
+
+    required_config.elasticsearch = Namespace()
+    required_config.elasticsearch.add_option(
+        'elasticsearch_class',
+        default='socorro.external.es.connection_context.ConnectionContext',
+        from_string_converter=class_converter,
+        reference_value_from='resource.elasticsearch',
+    )
+
+    #--------------------------------------------------------------------------
+    def save_raw_and_processed(
+        self,
+        raw_crash,
+        dumps,
+        processed_crash,
+        crash_id
+    ):
+        all_fields = SuperSearchFields(config=self.config).get()
```

Comment:
I do believe your proposed optimization, with hand-made caching, will be better than calling Elasticsearch every time we process a crash.